chromium_src/chrome/browser/policy/profile_policy_connector_builder.cc

```diff
@@ -0,0 +1,51 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "base/files/file_path.h"
+#include "chrome/browser/policy/profile_policy_connector.h"
+
+#define ProfilePolicyConnector BraveProfilePolicyConnector
+#define CreateProfilePolicyConnectorForBrowserContext \
+  CreateBraveProfilePolicyConnectorForBrowserContext
+#define CreateAndInitProfilePolicyConnector \
+  CreateAndInitBraveProfilePolicyConnector
+
+#include <chrome/browser/policy/profile_policy_connector_builder.cc>  // IWYU pragma: export
+
+#undef ProfilePolicyConnector
+#undef CreateProfilePolicyConnectorForBrowserContext
+#undef CreateAndInitProfilePolicyConnector
+
+namespace policy {
+
+std::unique_ptr<ProfilePolicyConnector>
+CreateProfilePolicyConnectorForBrowserContext(
+    SchemaRegistry* schema_registry,
+    CloudPolicyManager* cloud_policy_manager,
+    ConfigurationPolicyProvider* policy_provider,
+    policy::ChromeBrowserPolicyConnector* browser_policy_connector,
+    bool force_immediate_load,
+    content::BrowserContext* context) {
+  std::unique_ptr<BraveProfilePolicyConnector> connector =
+      CreateBraveProfilePolicyConnectorForBrowserContext(
+          schema_registry, cloud_policy_manager, policy_provider,
+          browser_policy_connector, force_immediate_load, context);
+  connector->GetBraveProfilePolicyProvider()->Init(schema_registry);
```

Comment:
why doesn't this happen in `CreateAndInitProfilePolicyConnector` when it calls `connector->Init`?